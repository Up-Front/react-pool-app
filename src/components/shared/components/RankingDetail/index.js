import React from 'react';
import PropTypes from 'prop-types';
import constants from './../../constants';

const RankingDetail = props => {
  const eloRating = props.currentRanking.eloRating || props.eloRating;
  const currentRanking = props.currentRanking.ranking;
  return (
    <div>
      {eloRating} {currentRanking}
    </div>
  );
};

RankingDetail.defaultProps = {
  eloRating: constants.DEFAULTELORATING,
  currentRanking: {},
  previousRanking: {},
};

RankingDetail.propTypes = {
  eloRating: PropTypes.number.isRequired,
  currentRanking: PropTypes.object.isRequired,
  previousRanking: PropTypes.object.isRequired,
};

export default RankingDetail;
