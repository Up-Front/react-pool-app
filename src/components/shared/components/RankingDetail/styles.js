import styled from 'styled-components';

export const RankingWrapper = styled.div`
  font-size: 24px;
  font-weight: bold;
`;

export const RankMovement = styled.span`
  position: relative;
  top: -3px;
  padding: 0 5px;
  font-size: ${props => props.theme.smallFont};
  font-weight: normal;
  color: rgba(255,255,255,0.5);
`;

export const RankMovementDirection = styled.span`
  position: relative;
  top: 4px;
  margin-right: 2px;
  display: inline-block;
  opacity: ${props =>
    props.movement === 0 || props.movement === undefined ? '0' : '1'};
  color: ${props =>
    props.movement > 0
      ? props.theme.movementUp
      : props.movement < 0 ? props.theme.movementDown : ''};

  transform-origin: center center;
  transform: ${props =>
    props.movement > 0
      ? 'rotate(-90deg)'
      : props.movement < 0 ? 'rotate(90deg)' : ''};

  font-size: 24px;
`;