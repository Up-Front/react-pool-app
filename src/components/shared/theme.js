// this is the theme for styled components colors
export const theme = {
  fontColor: '#FFF',
  fontSize: '16px',
  background: '#243949',
  borderColor: '#FFF',
  backgroundImage: 'linear-gradient(to top, #243949 70%, #517fa4 100%)',

  headerBackgroundColor: 'rgba(0, 0, 0, 0.7)',
  headerFontColor: '#FFF',

  buttonColor: 'DarkOrange',
  buttonFontColor: '#FFF',

  //match
  matchBackgroundColor: '#243949',
  contestedMatchBackgroundColor: 'red',

  //delete swipe
  deleteBackgroundColor: 'red',

  //user / competitor
  statusOnlineColor: 'green',
  statusOfflineColor: 'orange',
  statusUnknownColor: 'lightgrey',
};
