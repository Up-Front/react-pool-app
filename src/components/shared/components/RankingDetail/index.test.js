import React from 'react';
import { shallow } from 'enzyme';
import theme from './../../theme';
import RankingDetail from './';

describe('RankingDetail Component', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<RankingDetail them={theme} />);
    expect(wrapper).toMatchSnapshot();
  });
});
