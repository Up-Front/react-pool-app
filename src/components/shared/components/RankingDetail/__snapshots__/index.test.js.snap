// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RankingDetail Component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RankingDetail
    currentRanking={Object {}}
    previousRanking={Object {}}
    them={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        0,
        ".",
        <styled.span>
          <styled.span
                    movement={0}
          >
                    »
          </styled.span>
          -
</styled.span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      0,
      ".",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.span
              movement={0}
>
              »
</styled.span>,
            "-",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "»",
              "movement": 0,
            },
            "ref": null,
            "rendered": "»",
            "type": [Function],
          },
          "-",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          0,
          ".",
          <styled.span>
            <styled.span
                        movement={0}
            >
                        »
            </styled.span>
            -
</styled.span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        0,
        ".",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.span
                movement={0}
>
                »
</styled.span>,
              "-",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "»",
                "movement": 0,
              },
              "ref": null,
              "rendered": "»",
              "type": [Function],
            },
            "-",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
